Paper: Tell Me Something I Don't Know: Analyzing OkCupid Profiles, Haller, Narasimhan, and Shishido

juanshishido opened the pull request:
*Tell Me Something I Don't Know: Analyzing OkCupid Profiles*

We will continue adding and editing content.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+identifying deceit in online self-presentation and discuss how we're working to
+extend this analysis.
+
+This work was completed using SciPy, NumPy, pandas, Scikit-Learn, and NLTK.
```
This could probably be a reference/footnote

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+profiles.
+
+In this paper we analyze online self-presentation by combining natural language
+processing (NLP) with machine learning. We use this approach to motivate a
```
Consider specifying supervised or unsupervised learning.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+discussion of how to successfully mine text data. We describe our initial
+approach, explain why it was problematic, and discuss modifications that made
+it successful. In doing so, we review standard NLP techniques, challenge
+accepted "norms," and cover several ways to represent text data. We explain
```
norms needn't be in quotes.

I'm intrigued by the claims made in the abstract.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+approach, explain why it was problematic, and discuss modifications that made
+it successful. In doing so, we review standard NLP techniques, challenge
+accepted "norms," and cover several ways to represent text data. We explain
+clustering and topic modeling, including k-means and nonnegative matrix
```
This sentence is a bit awkward. `latter` refers to the last reference, which may mean topic modeling and/or non-negative matrix factorization (the last two points of the last two references), or something else.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+factorization, and discuss why we chose the latter. We then present unexpected
+similarities between self-identified drug users and those who declined to
+reveal their drug status. Finally, we discuss utilizing permutation testing for
+identifying deceit in online self-presentation and discuss how we're working to
```
Identifying deceit in online presentation is, by itself, an intriguing idea.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+------------
+
+Online dating has become a common way of finding mates, with 22% of 25-34 year
+olds having used online dating [Pew15]_. Previous studies suggest that the
```
Your comment here needs to examine more than just 24-34 year-olds. This cohort has had exposure to the Internet since high-school (or earlier) and would likely be early adopters of online dating. For this age demographic, I would've expected the number to be higher.

You should at least include overall demographics (what % of all people find mates online) to show salience across the entire population.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+surprisingly, research suggests that homophily extends to online dating, with
+people seeing mates similar to themselves [Fio05]_. However, it remains unclear
+if people within a particular demographic group self-present in a similar way
+when searching for a mate online.
```
When saying `it remains unclear if people within a particular demographic group...` are you particularly focusing on age? There are other kinds of demographic groups, so this claim is a bit generalized.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Description
+~~~~~~~~~~~
+
+The data was obtained from: https://github.com/everett-wetchler/okcupid
```
Recommend removing this sentence and moving the citation `[Wet15]_`  to `Profile information from 59,000+ OkCupid users [Wet15]_`.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+~~~~~~~~~~~
+
+The data was obtained from: https://github.com/everett-wetchler/okcupid
+[Wet15]_. Profile information from 59,000+ OkCupid users that were online on
```
The 1-2 sentences of this need to flow better, something like:

`We sampled profile information from 59,000+ OkCupid users [Wet15]_ that were online on 06/30/2012 who were within 25 miles of the San Francisco Bay Area.`

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+The data was obtained from: https://github.com/everett-wetchler/okcupid
+[Wet15]_. Profile information from 59,000+ OkCupid users that were online on
+06/30/2012. The users were within 25 miles of the San Francisco Bay Area. Each
+profile contains the user’s height, body type, religion, ethnicity, religion,
```
religion cited twice.

dalippa replied:
You were referring to age demographics earlier, and now break it down into multiple different demographics, excluding age. Please clarify some.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+..      Some custom LaTeX source here.
+
+References
```
Please use proper citation format that includes author, etc.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+First, we identified essays that we wanted to run analysis on. The essay text
+was cleaned by removing line break characters, urls, and stripping white
+spaces. If a user wrote less than 5 words for the given essay, s/he was removed
```
You can probably use gender-specific pronouns, or remove them by rewording. You can also alternate between he/she throughout the paper for gender balance.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Processing
+~~~~~~~~~~
+
+First, we identified essays that we wanted to run analysis on. The essay text
```
Consider rewording first sentence to avoid ending with a preposition, maybe `First, we identified essays for analysis`.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+First, we identified essays that we wanted to run analysis on. The essay text
+was cleaned by removing line break characters, urls, and stripping white
+spaces. If a user wrote less than 5 words for the given essay, s/he was removed
+from the analysis. Finally, stopwords and punctuation were removed.
```
You might want to define stopwords (or explain in a footnote).

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+from the analysis. Finally, stopwords and punctuation were removed.
+
+The essay was tokenized using the happyfuntokenizer
+(http://sentiment.christopherpotts.net/tokenizing.html), which is well suited
```
When tokenizing, did you also alter words of different tenses, plurality, etc? For example, would the tokens `eat, eats, eating` be considered a single token, or separate ones? When examining data in corpora, some tokenizers treat all three as the same.

ankurankan replied:
Link can be moved to citations.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+The essay was tokenized using the happyfuntokenizer
+(http://sentiment.christopherpotts.net/tokenizing.html), which is well suited
+for online communication as it maintains emoticons as discrete tokens. Finally,
```
Why would maintaining emoticons as discrete tokens be significant?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+The essay was tokenized using the happyfuntokenizer
+(http://sentiment.christopherpotts.net/tokenizing.html), which is well suited
+for online communication as it maintains emoticons as discrete tokens. Finally,
+a tfidf matrix was created with unigrams, bigrams and trigrams, dropping words
```
tfidf should probably be `tf-idf`. Please write out the full words and put the abbreviation in parentheses: ` term frequency–inverse document frequency (tf-idf)`

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+(http://sentiment.christopherpotts.net/tokenizing.html), which is well suited
+for online communication as it maintains emoticons as discrete tokens. Finally,
+a tfidf matrix was created with unigrams, bigrams and trigrams, dropping words
+with <0.01 document frequency.
```
Is this 1% or 0.01%?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+a tfidf matrix was created with unigrams, bigrams and trigrams, dropping words
+with <0.01 document frequency.
+
+To reduce cardinality, we combined demographic categories. For example,there
```
Did this affect your findings?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+Algorithms
+~~~~~~~~~~
+
```
You have 3 titles without any text. Do you need a section under methodology before delving into the algorithms?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+data set. Typically, we wish to find groups, or clusters, whose members are
+similar to each other.
+
+K-means splits a data set into k non-overlapping clusters. Each cluster is
```
Consider italicizing *k* to make it easier to read and identify.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+threshold.
+
+Choosing k and defining a similarity metric&mdash;recall that the goal with
+clustering is to find *similar* groups&mdash;are important considerations.
```
dash formatting issue.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+assignments no longer change or until the change is below some specified
+threshold.
+
+Choosing k and defining a similarity metric&mdash;recall that the goal with
```
dash formatting issue.

Choosing *k* is definitely important. Can you elaborate on why?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Choosing k and defining a similarity metric&mdash;recall that the goal with
+clustering is to find *similar* groups&mdash;are important considerations.
+
+NMF
```
Please write out abbreviations the first time you use them.

ankurankan replied:
This also isn't looking like a title in the PDF.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+finding the new mean value of each cluster. Finally, observations are
+reassigned to the nearest cluster centroid. This continues until cluster
+assignments no longer change or until the change is below some specified
+threshold.
```
Just wondering - did you use the SciPy kd-tree to assist with clustering?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+counts, so we calculated the tfidf matrix for unigrams, bigrams, and trigrams.
+We limited the tokens to those that appeared in at least 1% of the documents.
+We calculated NMF with k = 25, which factorized the tfidf matrix into two
+matrices, W and H. The dimensions are n_samples x 25 and 25 x n_features for W
```
When referring to math terms, please use the math macro so that it renders properly.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+and H, respectively. Group descriptions were given by top-ranked terms (the
+most distinctive) in the columns of H. Document membership weights were given
+by the rows of W. We calculated the maximum value in each row of W to determine
+essay group membership. We chose to have 25 groupings somewhat arbitrarily,
```
What would happen if you had arbitrarily chosen a different *k* (higher or lower)?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+previous studies of online dating [Nag09]_ [Tom12]_ [Bon05]_. However, a more
+recent study analyzing a much larger dataset [Sch13]_ reported that using an
+"open" vocabulary resulted in more generalizable results than the "closed" set
+available through LIWC. Instead of using LIWC, we created and selected our
```
Inconsistent acronym use: LIWC vs LWIC. Which is correct? Please write it out the first time used.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+available through LIWC. Instead of using LIWC, we created and selected our
+features using unigram, bigram, and trigram tokens from the combined text
+across the 10 essays for each user. The bigrams and trigram tokens were
+selected using their PMI, calculated as the frequency of the phrase divided by
```
What is PMI?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+selected using their PMI, calculated as the frequency of the phrase divided by
+the product of the individual words in that phrase. Bigrams were restricted to
+those with a PMI > 4 and trigrams were restricted to those with a PMI > 6, as
+described in Schwartz et al. (2013). We restricted all tokens to those that
```
Please use proper citation.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+After tokenizing the data, we applied PCA on the resulting features. The data
+was noisier than we anticipated, and required 50 principal components to
+account for 48% of the variance (Fig 1). PCA performed much worse when we
```
Please use Fig macro.

dalippa replied:
No figure uploaded.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+described in Schwartz et al. (2013). We restricted all tokens to those that
+appeared in at least 1% of the documents.
+
+After tokenizing the data, we applied PCA on the resulting features. The data
```
Please write out Principal Components Analysis the first time the acronym is used.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+After tokenizing the data, we applied PCA on the resulting features. The data
+was noisier than we anticipated, and required 50 principal components to
+account for 48% of the variance (Fig 1). PCA performed much worse when we
+scaled the data values. Whitening the data did not have an effect. We had also
```
What does it mean to whiten the data?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+calculates a ratio using the mean intra-cluster distance and the mean
+nearest-cluster distance for each sample. Because the dataset was so large, we
+calculated the score on subsets of the data, creating a bootstrap estimate of
+the score. We applied k-means clustering with 3, 5, 7, and 9 clusters, and
```
Why these numbers of clusters? Was this also arbitrary?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Yet even with the additional preprocessing, dimensionality reduction, and
+clustering, the groupings were sparse and uneven. We decided that PCA and
+k-means clustering were probably not well-suited for our dataset. We attempted
+using k-modes (from: https://github.com/nicodv/kmodes) instead of k-means, and
```
Please move link to citation.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+clustering, the groupings were sparse and uneven. We decided that PCA and
+k-means clustering were probably not well-suited for our dataset. We attempted
+using k-modes (from: https://github.com/nicodv/kmodes) instead of k-means, and
+although the clusters were more evenly distributed (Fig 2), the costs were
```
Use Figure macro.

dalippa replied:
No figure uploaded.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+~~~~~~~~~~~~~~
+
+The initial essay preprocessing contributed a significant amount of noise to
+our analysis. Specifically, while preprocessing, we combined all essays into a
```
How/why did pre-processing introduce noise? This might be salient for others doing similar styles of processing.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+our analysis. Specifically, while preprocessing, we combined all essays into a
+single text block for each user. In grouping all essays together introduced
+unnecessary noise into our analysis. Instead, we decided to focus our analysis
+on two separate essays: "My Self Summary" and "Favorite Books, Movies, TV."
```
Why did you choose these two essays?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+on two separate essays: "My Self Summary" and "Favorite Books, Movies, TV."
+
+We began by exploring the lexical features of the text. Our goal was to
+determine whether there existed inherent differences in writing styles by
```
The ordering of this paragraph is a bit odd. Consider moving the `Our goal...` as the first sentence to improve clarity.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+When determining essay length, we used our tokenizer, as opposed to, say,
+simply splitting the text on white-space. This was mainly to be consistent with
+our downstream analyses, such as predictive modeling.
```
Again, to my earlier question, did the tokenizer treat all forms of a word as the same? or differently? (such as eat, eats, eating, eaten)? Did it normalize nouns and treat them as single tokens, and treat verbs separately?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+A list of profane words was obtained from the "Comprehensive Perl Archive
+Network" website. Slang terms include words such as "dough," which refers to
+money, and acronyms like "LMAO," which stands for "laughing my ass off." These
+terms come from the Wiktionary Category:Slang page. Note that there is overlap
```
Instead of `These terms come from the Wiktionary...` please use a citation.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+users tend to use verbs ("action" words) more often than other groups of users?
+To answer questions like these, we first had to associate parts of speech (also
+known as "lexical categories") with each term (or "token") in our corpus. To do
+this, we used spaCy's part-of-speech tagger. We use spaCy's "coarse-grained"
```
Please add citation for spaCy.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+known as "lexical categories") with each term (or "token") in our corpus. To do
+this, we used spaCy's part-of-speech tagger. We use spaCy's "coarse-grained"
+tags, of which there are 19, in order to maintain low-cardinality. These tags
+expand upon Petrov, Das, and McDonald's "universal part-of-speech tagset."
```
Please add citation

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+In addition to lexical characteristics, we were interested in understanding
+the semantics of the text. To do this, we used non-negative matrix
+factorization (NMF) in order to find latent (or hidden) structure in the text.
```
you can remove "(or hidden)"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+each resulting group. The keyword algorithm takes the 1000 most frequent
+unigrams and extracts hypernyms from them using WordNet. After the hypernyms
+are calculated, it uses examples of these hypernyms as seeds to find contextual
+4-grams. It then filters the 4-grams to keep only those that occur more than 20
```
Is it more consistent to use 1-gram and 2-gram for the unigram and bigram? Or is this standard notation?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Results
+-------
+
+In this section, we describe our findings. We start with a discussion of our
```
First sentence is redundant (due to title). Consider moving the analysis to a separate section.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+lexical-based analyses before discussing our semantic-based results.
+Lexical-based characteristics include essay length, use of profanity and slang
+terms, as well as part-of-speech usage. Our analyses focus on two demographic
+dimensions—sex and drug usage—and on two essays—"my self summary" and "favorite
```
Reverse the terms `sex and drug usage` to `drug usage and sex` for clarity.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+books, movies, shows, music, food."
+
+We first compare lexical-based characteristics on the self-summary text by sex.
+Our sample includes 21,321 females and 31,637 females. Note that the difference
```
which set were males?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+We first compare lexical-based characteristics on the self-summary text by sex.
+Our sample includes 21,321 females and 31,637 females. Note that the difference
+between this and the 59,946 users in the data set is due to our dropping users
```
Can be reworded more concisely.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Our sample includes 21,321 females and 31,637 females. Note that the difference
+between this and the 59,946 users in the data set is due to our dropping users
+with less than five tokens in a particular essay. We find that, on average,
+females write just under 150 words (tokens, actually) compared to males' 139,
```
Why use an inexact number ("just under 150") for females and an exact number for males?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+with less than five tokens in a particular essay. We find that, on average,
+females write just under 150 words (tokens, actually) compared to males' 139,
+though the variance is higher for the males than the females. This difference
+is statistically significant.
```
Proof?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+   | Nouns          | 18.65% | 18.86% | *           |
+   +----------------+--------+--------+-------------+
+   | Verbs          | 18.28% | 18.27% |             |
+   +----------------+--------+--------+-------------+
```
Caption for table? Consider changing the asterisks and instead consider putting "for males", or "for females".

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Not surprisingly, slang is much more prevalent (on a per-user basis) than
+profanity. 56% of users use some form of slang in their self-summary essays.
+Females use slang at a lower rate than males—54% vs. 57%—a difference that is
+statistically significant.
```
Why is the slang difference statistically significant? Do you have substantiation?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+Our first attempt at visualizing tokens that make the groupings distinctive was
+to plot a histogram of the difference in relative frequency between tokens for
+a given group versus other groups (Fig 3). This can be done between clusters
```
Missing figure.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+people in that particular demographic group. We visually examined the clusters
+and noted the ones dominated by a particular demographic group. For example,
+in the "Favorite books, movie, TV" essay, we noted that group 7 was dominated
+by people identifying as black relative to other ethnic groups (Fig 5, left).
```
Missing figure

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+the theme associated that cluster.
+
+We were also able to split and combine demographic groups, for example gender
+and sexual orientation (Fig 6). For the “Self Summary” essay we noted that
```
missing figure; cited sexual orientation as a demographic here, but not earlier.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+demographic splits we noted that cluster 1 discussed San Francisco and the act
+of moving or locations and changing. The word cloud for group 6 was represented
+by n-grams discussing searching for a relationship and types of relationships.
+This is in contrast to cluster 2, which was dominated by women and whose word
```
This section is difficult to review without the corresponding figure.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+users. This was most distinctive when the separation between drug users and
+non-drug users was large (e.g., groups 3, 7, 21, and others). Our hypothesis is
+that it is possible that individuals who do not respond to the drug usage
+question might not want to admit that they use drugs.
```
Paper claimed identifying deceit without explicitly showing that as a conclusion, and how it was achieved.

dalippa commented:
Overall comments:
* Excellent premise, and fascinating analysis
* For the most part, well written and clear

Major issues that need to be fixed:
* Missing sections: results, future work, conclusion, acknowledgments
* 1 significant unsubstantiated claim (`identifying deceit in online presentation and how we're working to extend this analysis`)
* Major choices/decisions (cluster sizes, essay choice) should be explained more.
* Missing cited figures (makes the last few paragraphs difficult to review)
* Missing and improper citations

Minor issues that should be fixed:
* Some questions, clarity issues
* Writing out acronyms the first time used (for clarity)
* Minor formatting, citation issues (such as using the correct formatting macro)
* Occasionally verbose

juanshishido commented:
@dalippa: Thanks for providing such detailed comments. We really appreciate the feedback! It's quite helpful.

We're currently working on adding more text, making figures consistent (style-wise), and tweaking a few additional things. We will also go through each of your comments and update accordingly.

Thanks again!

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+.. class:: keywords
+
+   natural language processing, machine learning, okcupid, online dating
```
Consider adding supervised/unsupervised learning as a keyword.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+
+.. [Nag09] Nagarajan and Hearst, An Examination of Language Use in Online Dating Profiles, 2009
+
+.. [Wet15] Everett Wetchler, okcupid, (2015), GitHub repository,
```
It might also be helpful to specify why you chose OkCupid as your data source. Some of your conclusions might not be evident in other online dating profiles, but be an artifact of how their specific profiles are constructed. I would consider adding in "Future Work" something about sampling data from other sources.

ankurankan commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,424 @@
+Algorithms
+~~~~~~~~~~
+
+K-means
```
This doesn't show as a title in the generated PDF.

ankurankan commented:
Looks like I am too late to review :smile: . @dalippa has covered everything.

sbenthall commented:
Build server is currently reporting a Malformed Table build error for this paper.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+   |                | exploring loves            |                            |
+   +----------------+----------------------------+----------------------------+
+
+We use NMF to help us understand the subject matter that users find interesting
```
The first sentence is a bit awkward, maybe: `We use NMF to help us understand the subject matter that users find interesting about themselves and share in their public profile.`

The second sentence isn't really needed.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+and important about themselves and, thus, choose to write about. This provides
+insight into the way they choose to self-present. In addition to particular
+themes, NMF also allows us to consider stylistic expression. Choosing the
+number of NMF components—these can be thought of as topics to which users are
```
This sentence is really awkward since the inner sentence is an aside to the outer one.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+clustered—is an arbitrary and iterative process. For the self-summary essay, we
+chose to start with 25.
+
+Several expected themes emerged. Some users, for example, chose to highlight
```
Specify a summary of the themes before diving into them, such as: `Several themes quickly emerged, but the themes were as varied as the personalities of the authors: highlighting personality traits or activities; using colloquialisms like "easy going"; and others were quite direct about what they were seeking.` Then you might be able to delve into large cohorts that fit the patterns.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+   | straight talk  | know, just, want, ask, message, just ask, really,       |
+   |                | talk, write, questions                                  |
+   +----------------+---------------------------------------------------------+
+   | about me       | 'm pretty, 'm really, 'm looking, 'm just, say 'm,      |
```
Is `'m really` really a token? I remembered reading something about how punctuation was handled with tokenizing.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+were looking for, whether that be a long-term relationship, a friendship, or
+sex. Topics and a selection of their highest weighted tokens are summarized in
+the table below. (Note that a complete list of the 50 highest weighted tokens
+for each topic is available in the appendix.)
```
Thank you for putting the 50 highest in the appendix, but the number here is quite long too. Consider abbreviating it.

Since you have the entire table in the appendix, you could also show the most interesting items in the table, and have the one in the appendix ordered by weight.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+traveling, and cooking were popular choices. Others chose to mention what they
+were looking for, whether that be a long-term relationship, a friendship, or
+sex. Topics and a selection of their highest weighted tokens are summarized in
+the table below. (Note that a complete list of the 50 highest weighted tokens
```
Instead of using a sentence to highlight what is in the appendix, you could rephrase with links:

Table X contains ______ topics and their associated tokens; for a complete list, see [Appendix A].

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+
+In order to determine whether there are differences in the topics or themes
+that OkCupid users choose to write about in their self-summaries, we plot the
+distribution over topics by demographic split. This allows us to identify how
```
Be explicit with the demographic split with the plots - especially since I am not seeing any graphics.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,604 @@
+Permutation Testing
+*******************
+
+Permutation testing ( WE DIDN"T END UP DOING THIS - SKIP?)
```
If it wasn't done, please remove.

dalippa commented:
@juanshishido You are most welcome. After seeing more activity, I thought it might be ready for re-review and started commenting again. Sorry if I refer to areas that are still works in progress.

Please notify me when you're ready for a re-review.

Thanks!

sbenthall commented:
We are getting the following error on our build server.

https://gist.github.com/sbenthall/726a597f20388ef74bf1229e93f04c07

sarostru commented:
The build server is having trouble with one of your tables, 
```
• Build the paper...
:345: (ERROR/3) Malformed table.
+----------------+---------------------------------------------------------+
| Topic | Tokens |
+================+=========================================================+
| meet & greet | meet new people, looking meet new, love meeting new, |
| | new friends, enjoy meeting, interesting people, |
| | want meet, 'm new, people love, experiences |
+----------------+---------------------------------------------------------+
```

juanshishido commented:
Thanks for the head's up, @sarostru. I noticed and fixed a misaligned `|`.

I was wondering how really long tables would render in the PDF. Hopefully it's okay.

It says the build for paper 8 is scheduled, so we'll see.

juanshishido commented:
@sarostru: looks like that fixed it. Thanks!

jseabold commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,748 @@
+matrix using spaCy's default tokenizer with punctuation removed. We chose to
+include unigrams, bigrams, and trigrams [5]_. Stop words [6]_ and terms that
+appeared in less than 0.5% of documents were removed. Stemming, the process of
+of removing word affixes, was not done.
```
two `of`s

jseabold replied:
Might also be nice to give a little explanation about why you are not doing stemming here.

jseabold commented:
Nice and interesting paper. A few minor things. I would like to see forma citations for the software packages used. Where this isn't available (spaCy AFAICT) a footnote linking to their home page would be nice.

Style-wise. I'm not sure if there's a preference for endnotes or footnotes. I couldn't find anything in the proceedings. 

AFAICT, you never mention your final vocabulary sizes for the essays after pre-processing and tokenization. Would be nice to see these. Maybe even a distribution of these.

dalippa commented:
@jseabold when I had submitted my paper last year, the consensus with my review group were that footnotes can be used for details/clarification and endnotes were mostly for references (to improve readability).

juanshishido commented:
@dalippa, @ankurankan, @sbenthall, @sarostru, @jseabold: We would like to thank all of you for taking the time to review and provide such useful and detailed comments (and for alerting us to build errors!). The feedback was incredibly helpful.

juanshishido commented:
Also, I just noticed that I forgot to remove two inline comments. Fixed (plus one spelling error) with 861bea5. None of the other content has changed. I hope this is okay.

juanshishido commented:
@jseabold: We can easily add formal citations for spaCy (scikit-learn, too?). Good catch on the vocabulary sizes! We have that information in a Jupyter notebook in our analysis repo. We can add that, too. Shall I make those changes?

dalippa commented:
@juanshishido Links to the Jupyter notebook from your repository would be a great addition.

juanshishido commented:
Thanks, @dalippa! Added.

Still waiting to hear if making those other changes is okay.

darribas commented:
Sorry to be late to the party, which makes it easier for me given the excellent comments already made!

A couple of additional minor things:

* Agree w/ @jseabold , it'd be great to have actual citations for software package when available.
* Consider including a small equation when explaining `tf-idf`, sometimes it clarifies the language.
* I can see how it's interesting in itself, but I think it'd be good to have a couple of lines of why you chose to focus on drug usage and sex (gender).
* P. 2 has "significantly significant". I think you mean "statistically significant".
* I'd recommend to explicitly refer to the figure ("Figure 1..." as oposed to "The following figure").
* Consider re-arranging the classes in the graphs in a way they are sorted by users in each cluster instead of alphabetically. I think it'd make the graph a bit more useful.
* I'd love to see a table of the marginal effects and standard errors of the word usage in the regression. That'd directly give insight into how each word affects the probability of drug usage.

Some of those are comments/suggestions rather than things to definitely change. Once the basic ones are fixed, the paper is good to go AFAIC.

dalippa commented:
I hadn't time to review your changes yet. I'll take a look later tonight.

On Thursday, July 7, 2016, Juan Shishido <notifications@github.com> wrote:

> Thanks, @dalippa <https://github.com/dalippa>! Added.
>
> Still waiting to hear if making those other changes is okay.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/207#issuecomment-231274413>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHdFxY4VqwX9hcaxZXCfufeewzu2lCeKks5qTdeFgaJpZM4IqMFl>
> .
>


-- 
Please forgive any typos since iSent this message from my iPad.

sarostru commented:
Due to the complication of the proceedings deadline lining up with everyone's travel schedule to attend the conference we've decided to allow authors to spend an additional 2 days, that is until July 12th to complete any major changes to the submission.  This may impact your enjoyment of the conference, so we hope not too many of you need to take the time.  For reviewers, we ask that you read over this final document before the conference is over and post your decision here in the comments on or before July 17th.

Thanks and have a great Scipy!
Your proceedings co-chairs Scott Rostrup (@sarostru) and Sebastian Benthall (@sbenthall)

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+------------
+
+Online dating has become a common and acceptable way of finding mates. 
+Almost half of Americans know someone who uses
```
Very Minor: This may be picking nits, but it might be good to specify if it's just over half or under half (to contrast with the 59% figure provided later in the sentence).

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Online dating has become a common and acceptable way of finding mates. 
+Almost half of Americans know someone who uses
+or who has met a partner through online dating and 59% believe online dating is
+a good way to meet people [Pew16]_. Online dating sites or mobile dating apps
```
The paper is also salient because online dating is an increasing trend among younger demographics, then cite this.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+are used by 27 percent of 18-24 year olds, 22 percent of 25-34 year olds, 21
+percent of 35-44 year olds, 13 percent of 45-54 year olds, and 12 percent of
+55-64 year olds [Pew16]_. Given the popularity of online dating, the way that people 
+self-present online has broad implications for the relationships they pursue. 
```
Self-presentation online has wider implications, such as social media (maybe cited in future work)

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+55-64 year olds [Pew16]_. Given the popularity of online dating, the way that people 
+self-present online has broad implications for the relationships they pursue. 
+
+Previous studies suggest that the free-text portion of online dating profiles
```
Minor, for better flow: "After photographs, previous studies suggest..."

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Previous studies suggest that the free-text portion of online dating profiles
+is an important factor (after photographs) for assessing attractiveness
+[Fio08]_. The principle of homophily posits that people tend to
+associate and bond with similar others and that this principle strongly
```
awkward: "with similar others"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+structures social networks and ties, most prominently by race and ethnicity
+[McP01]_. Perhaps not surprisingly, research suggests that homophily extends to
+online dating, with people seeking mates similar to themselves [Fio05]_.
+However, it remains unclear whether people within particular demographic groups,
```
"demographic groups" -> demographics

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Specifically, we focus on whether people signal demographic characteristics
+through the way they present themselves online. We extend previous natural
+language processing analyses of online dating [Nag09]_ by combining NLP
+with supervised and unsupervised machine learning on a larger scale. We
```
larger scale than what? [nag09]? Could you clarify more?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Description
+~~~~~~~~~~~
+
+The data was obtained from Albert Y. Kim's JSE_OkCupid repository [2]_. Profile
```
First sentence isn't needed. You can change to `Profile information [2]_ was available for...`

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+The data was obtained from Albert Y. Kim's JSE_OkCupid repository [2]_. Profile
+information was available for 59,946 OkCupid users that were members as of
+06/26/2012, lived within 25 miles of San Francisco, had been active in the
```
maybe: active -> active members

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+The present study extends previous work by analyzing additional features and by
+introducing novel analyses.
+
+This data set is public, as authorized by OkCupid president and co-founder
```
This should probably move to [2]_ which is the citation for the dataset.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Methods
+-------
+
+Term Frequency-Inverse Document Frequency
```
Consider adding the acronym TF-IDF in parentheses here.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+Line break characters and URLs were removed from the essay text. Multiple
+periods, dashes, and white spaces were replaced by single instances. Essays
+were tokenized (segmented) into individual words using spaCy's default
```
you can leave out tokenized and use segmented, since you reference the tokenizer term later.

Also, citation for spaCy's default tokenizer.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+emoticons as discrete tokens, and removes punctuation. Users who wrote less
+than 5 words for a given essay were removed from the analysis.
+
+We combined drug usage status levels. Specifically, users who responded
```
You may want to specify *why* you modified drug usage answers.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+Tf-idf is the product of the term frequency and the inverse document frequency.
+The term frequency refers to the *relative* frequency of term :math:`t` in
+document :math:`d`. The inverse document frequency is the log of the total
```
does the logarithm base matter?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For document clustering, the document corpus is projected onto a
+k-dimensional semantic space, with each axis corresponding to a particular
```
`k` should be italicized.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+semantic space to be orthogonal, and therefore is able to find directions for
+related or overlapping topics.
+
+NMF was applied to each essay of interest using scikit-learn (version 0.17.1),
```
Version number may not be required here. Consider a reference here to scikit-learn, including version number.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+(minimum frequency). NMF was calculated with :math:`k` dimensions, which
+factorized the tf-idf matrix into two matrices, :math:`W` and :math:`H`. The
+dimensions were ``n_samples x k`` and ``k x n_features`` for :math:`W` and
+:math:`H`, respectively. Group descriptions were given by top-ranked terms (the
```
"the most distinctive" is unnecessary.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Approach
+--------
+
+Our analyses focus on two demographic dimensions — sex and drug usage — and on
```
the PDF formatter might need `emdash` LaTeX macro to show correctly; as is, it's in bold.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+two essays — "My self summary" and "Favorite books, movies, shows, music, food."
+These essays were selected because they were answered by most users. For
+example, "The most private thing I am willing to admit" prompt was ignored by
+32 percent of users. Other essays in this data set may provide additional
```
use % and percent consistently.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+part-of-speech tagger. spaCy supports 19 coarse-grained tags [4]_ that expand
+upon Petrov, Das, and McDonald's universal part-of-speech tagset [Pet11]_.
+
+Differences in lexical features by demographic were analyzed using permutation
```
You don't mention *why* you chose this method (comparing by gender first, particularly with respect to profanity). Was there prior work that influenced this?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+identified the most distinctive terms in each lexical category by sex using the
+smoothed log-odds-ratio, which accounts for variance.
+
+Text semantics were also analyzed. The corpus was transformed into a tf-idf
```
Consider: "We also analyzed text semantics by transforming the corpus into a tf-idf..."

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+previous year, and had at least one photo in their profile [Wet15]_.
+The data set contained free-text responses to 10 essay prompts as well as the
+following user characteristics: age, body type, diet, drinking status, drug
+usage status, education level, ethnicity, height, income, job type, location,
```
You state education as a user characteristic but do not show its analysis.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+In addition to part-of-speech usage, we explored specific terms associated
+with parts-of-speech that were distinctive to a particular sex. We did this
+using the log-odds-ratio. The 10 most-distinctive adjective, noun, and verb
```
Consider moving "The 10 most-distinctive..." to a caption under the table.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+tokens for each sex are summarized below.
+
+   +----------------+----------------------------+----------------------------+
+   | Part-of-Speech | Female                     | Male                       |
```
"Part of Speech"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+   |                | friends                    | women video technology     |
+   |                |                            | geek                       |
+   +----------------+----------------------------+----------------------------+
+   | Verbs          | love am laugh laughing     | m was play playing laid    |
```
`m` and `'ll`?

dalippa replied:
@matarhaller @juanshishido Please clarify on the tokens `'ll` and `m` here.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+NMF was then used to provide insights into the underlying topics that users chose to
+describe themselves. Selecting the number of NMF components (topics to which
+users are clustered) is an arbitrary and iterative process. For the
+self-summary essay, we chose 25 components, which resulted in a diverse, but
```
"diverse, but manageable,"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+Several expected themes emerged. Many users chose to highlight personality
+traits, for example "humor" or "easy-going," while others focused on describing
+the types of activities they enjoyed. Hiking, traveling, and cooking were
+popular choices. Others chose to mention what they were looking, whether that
```
Others chose to mention what they were looking" (missing preposition) -> "Others chose to mention what kind of relationship they were seeking"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+We further examined online self-presentation by considering the other available
+essays in the OkCupid data set. Previous psychology research suggests that
+people's preferred music styles are tied to their personalities [Col15]_, and it
```
"people's" -> " a person's"

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+   | everything     | books, movies, food, music, shows, country, dance,      |
+   |                | action, lots, horror                                    |
+   +----------------+---------------------------------------------------------+
+   | movies-drama-0 | eternal sunshine, spotless mind, litte miss sunshine,   |
```
What is the difference between `movies-drama-0` and `movies-drama-1`? Why were they split out?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+The topics for this essay were less distinctive than the topics for the
+self-summaries. In some cases, genres (or media) overlapped. For example, the
+"TV-comedies-0" group included "The Walking Dead," which is a drama. There was
+also overlap between groups. This might suggest that the number of NMF
```
Try to remove words like "might". What *does* this suggest?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+   Favorites
+
+Although we were unable to test this hypothesis directly due to lack of
+ground-truth drug-usage status for these users, the manner by which free-text
```
"ground-truth" -> the true

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+   Favorites
+
+Although we were unable to test this hypothesis directly due to lack of
```
Does this paragraph belong in the drug usage section, or in the Favorites section?

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+and split the data into training and test sets.
+
+The full-text model accuracy increased to 72.7 percent. We used the feature
+weights to find the 25 most-predictive drug-usage terms. These are listed below.
```
"... predictive drug-usage terms listed below" (and remove the last sentence)

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+
+The current study extended previous NLP analyses of online dating profiles. The
+scope of this work was larger than previous studies, both because of the size
+of the datasets (previous studies used much smaller datasets) and because of
```
I don't remember you mentioning dataset size (with previous studies) earlier in the paper. That's salient and should be there as well.

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+scope of this work was larger than previous studies, both because of the size
+of the datasets (previous studies used much smaller datasets) and because of
+the novel combination of NLP with both supervised and unsupervised machine
+learning techniques (NMF and logistic regression). To our knowledge, there is currently no study that
```
Consider: "supervised and unsupervised machine learning techniques, such as NMF and logistic regression".

dalippa commented on papers/juan_shishido/juan_shishido.rst:
```diff
@@ -0,0 +1,750 @@
+comments and suggestions were invaluable, helping us organize our
+thoughts and analyses.
+
+.. Customised LaTeX packages
```
This can be removed.

dalippa commented:
@juanshishido You've substantially improved the paper across the board, with the graphics and fleshing out all of the sections. I recommend accepting the paper to the proceedings, with the following additional revisions (for details, see comments inline):

Major issues:
* Missing specifics in the conclusion. Two in particular is enthusiasm in female self-presentation; the other, drug usage findings.
* No future work section. Rather than citing future ideas throughout the paper, dedicate 1 paragraph to exploring the next step in this research (unintentionally providing information; online deception). One area of future work that was completely missing was the extension of these principles to social media profiles, particularly Facebook (for similar personal reasons) and LinkedIn (for analyzing professional profiles).

Minor issues:
* Minor clarification, organizational and grammar errors
* Minor readability issues due to overuse of parentheses (consider footnotes, rewording, etc to improve readability)
* Some clarification needed on the 25 categories in the analysis (such as movies-drama-0 vs movies-drama-1)
* Extensive use of passive voice (active voice is preferred when possible for better readability)
* Extensive tense switches. Consider a unified tense (past or present) for a better narrative.
* Some characteristics, such as education, almost certainly play a role in the profile characteristics. Was this excluded from the paper, or an area of future study?
* Possible tokenizer issue with `'m` cited. Please clarify if this is intentional or an error.

juanshishido commented:
Thanks for the information regarding the new July 12, @sarostru and @sbenthall!

Starting to work through some of the comments now.

juanshishido commented:
Thanks so much, @dalippa!! We benefitted greatly from your thorough and thoughtful comments. We are quite grateful.

We are continuing to make the suggested edits. I almost have another commit ready. We'll plan to add more Monday night and Tuesday.

Hope to meet all of you in Austin!

Thanks again!

michaelpacer commented:
Here's some advice from Proselint (http://github.com/amperser/Proselint), a Python-based linter for prose, with six false alarms removed (23 instances, including 19 instances of "" vs “”):

    151:5: garner.phrasal_adjectives.ly No hyphen is necessary in phrasal adjectives with an adverb ending in -ly.
    132:-7: typography.symbols.sentence_spacing More than two spaces after the period; use 1 or 2. Found once elsewhere.
    269:46: weasel_words.very Substitute 'damn' every time you're inclined to write 'very;' your editor will delete it and the

sarostru commented:
@dalippa @ankurankan @darribas @jseabold , I think I got everyone who read the paper, from the comments, I think the status is ready with some updates which are ongoing.  If true, then please go ahead and add the ready label.

@juanshishido @matarhaller once you are done making the updates please do a final proofread and @sbenthall and I will help with updating the proceedings build from draft to final.

darribas commented:
Ready to go for me!

dalippa commented:
@sarostru @sbenthall @juanshishido @matarhaller 

Two final comments:
* Please clarify on the token issues (`m`, `'ll`)
* Remove qualifiers like "might". If the data is inconclusive, say so explicitly. If it's conclusive, remove the qualifier.

After these two comments are addressed, I'm good with marking the paper as ready.

juanshishido commented:
We've addressed the token issues and removed the "might" qualifiers.

We will do a final proofread and make any final edits in the next few days. Thanks!
